### PR TITLE
Add wgsl support for FromPixels

### PR DESCRIPTION
**Pull request by shaoboyan**
This PR add wgsl shaders for FromPixels copyExternal
version. And also upgrade @webgpu/types to have latest
type names.

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5552)

### CODE REVIEW

**Comment by shaoboyan** [comment]
@qjia7  PTAL, thanks!

**Comment by shaoboyan** [comment]
@qjia7 Updated, PTAL again, thanks!

File: tfjs-backend-webgpu/src/kernels/FromPixels_utils/from_pixels_webgpu.ts [hunk 1]
```diff
@@ -169,17 +199,17 @@ export class FromPixelsProgram implements WebGPUProgram {
...
-      visibility: GPUShaderStage.COMPUTE,
-      storageTexture: {access: 'read-only', format: 'rgba8unorm'}
-    });
+    this.useWgsl ?
+        bindGroupLayoutEntries.push(
+            {binding: 1, visibility: GPUShaderStage.COMPUTE, texture: {}}) :
+        bindGroupLayoutEntries.push({
+          binding: 1,
+          visibility: GPUShaderStage.COMPUTE,
+          storageTexture: {access: 'read-only', format: 'rgba8unorm'}
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/kernels/FromPixels_utils/from_pixels_webgpu.ts:208]: Does it mean that `imageLoad` corresponds to `storageTexture` not `texture` in webgpu?

**Review by qjia7 — APPROVED** [review]
@kainino0x @lina128 take a look, thanks.

**Comment by kainino0x** [comment]
> @kainino0x @lina128 take a look, thanks.

Do you have access to add me as a reviewer on GitHub rather than @ mentioning me? If you do that it will go into my "review" email label and I'll be able to see it more promptly.

**Comment by qjia7** [comment]
> > @kainino0x @lina128 take a look, thanks.
> 
> Do you have access to add me as a reviewer on GitHub rather than @ mentioning me? If you do that it will go into my "review" email label and I'll be able to see it more promptly.

Yes. Usually I will do both. Sorry I just forgot for this one.

**Comment by kainino0x** [comment]
> Yes. Usually I will do both. Sorry I just forgot for this one.

Ah no problem!

File: tfjs-backend-webgpu/src/kernels/FromPixels_utils/from_pixels_webgpu.ts (hunk 1 above)
Comment (kainino0x) [reply tfjs-backend-webgpu/src/kernels/FromPixels_utils/from_pixels_webgpu.ts:208]: In chromium textureLoad should be usable with either storage or non-storage textures.

However note in the spec, reading from storage textures has been removed entirely:
https://github.com/gpuweb/gpuweb/pull/1794
... until we have feedback that they are needed.
Please provide feedback if you're unable to use a non-storage texture binding here.

**Review by kainino0x — APPROVED** [review]

File: tfjs-backend-webgpu/src/kernels/FromPixels_utils/from_pixels_webgpu.ts (hunk 1 above)
Comment (shaoboyan) [reply tfjs-backend-webgpu/src/kernels/FromPixels_utils/from_pixels_webgpu.ts:208]: > In chromium textureLoad should be usable with either storage or non-storage textures.

Yes, why we have this PR is due to glsl version shader reports warning about using deprecate built-in function. And I find that readonly storage texture is not supported in wgsl spec. So I use non-storage texture here.  
the tests shows that change to use non-storage texture works fine here.

File: tfjs-backend-webgpu/src/kernels/FromPixels_utils/from_pixels_webgpu.ts [hunk 2]
```diff
@@ -55,6 +57,28 @@ export class FromPixelsProgram implements WebGPUProgram {
 
   constructor() {
     this.shaderKey = 'fromPixels';
+    this.useWgsl = true;  // getUseWgsl();
+  }
+
+  getUserCodeWgsl(): string {
+    const userCode = `
+    [[binding(1), group(0)]] var src: texture_2d<f32>;
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/kernels/FromPixels_utils/from_pixels_webgpu.ts:65]: Is above line the only change between from_pixels_webgpu and from_pixels_import_webgpu?  If yes, I prefer you reuse the shader code.
For example, define a common function like below:
```
export function makeFromPixelsSource()
{
  return `
    ${getMainHeaderStringWgsl(this.workGroupSize)} {
      ${getGlobalIndexStringWgsl(this.workGroupSize)}
      let flatIndexBase = index * uniforms.numChannels;
      let coords: vec3<u32> = getCoordsFromFlatIndex(flatIndexBase);
      let values = textureLoad(src, vec2<i32>(coords.yx), 0);
      for (var i: u32 = 0u; i < uniforms.numChannels; i = i + 1u) {
        var value = values[i];
        var flatIndex = flatIndexBase + i;
        if (flatIndex < uniforms.size) {
          result.numbers[flatIndex] = i32(floor(255.0 * value));
        }
      }
    }
`;
}

getUserCodeWgsl(): string {
    const userCode = `
    [[binding(1), group(0)]] var src: texture_2d<f32>;
    ${makeFromPixelsSource()}
   `;
}

Comment (shaoboyan) [reply tfjs-backend-webgpu/src/kernels/FromPixels_utils/from_pixels_webgpu.ts]: Good cleanup! Done.

**Review by qjia7 — APPROVED** [review]
LGTM, thanks.

**Review by lina128 — APPROVED** [review]
LGTM

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5552#-:-MiS5i48EI7du9CAk16i:bb6ywmx)* status: :shipit: complete! 3 of 1 approvals obtained (waiting on @qjia7)

**Pull request merged**